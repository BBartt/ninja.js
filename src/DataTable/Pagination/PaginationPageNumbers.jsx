import React from 'react';

const PaginationPageNumbers = ({ pageNumber, currentPageNumber, onClick }) => {
  return (
    <li className="page-item mr-1">
      <button
        className={`page-link${
          currentPageNumber === pageNumber ? ' button-outline' : ''
        }`}
        onClick={() => onClick(pageNumber)}
      >
        {pageNumber + 1}
      </button>
    </li>
  );
};

export default PaginationPageNumbers;
