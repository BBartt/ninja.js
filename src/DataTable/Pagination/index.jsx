import React from 'react';

import PaginationPageNumbers from './PaginationPageNumbers';

const Pagination = ({ currentPageNumber, totalNumberOfPages, onClick }) => {
  const pages = [...Array(totalNumberOfPages).keys()].map(pageNumber => (
    <PaginationPageNumbers
      currentPageNumber={currentPageNumber}
      onClick={num => onClick(num)}
      pageNumber={pageNumber}
      key={pageNumber}
    />
  ));

  if (pages.length <= 1) return null;

  return <ul className="pagination">{pages}</ul>;
};

export default Pagination;
