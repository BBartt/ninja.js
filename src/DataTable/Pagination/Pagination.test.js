import React from 'react';

import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Pagination from './';

import { dummyData } from '../../dummyData/dummyData';

describe('<Pagination />', () => {
  it('renders without crashing', () => {
    shallow(
      <Pagination
        currentPageNumber={1}
        totalNumberOfPages={10}
        onClick={() => {}}
      />
    );
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Pagination
          currentPageNumber={1}
          totalNumberOfPages={10}
          onClick={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders 10 PaginationPageNumbers', () => {
    const wrapper = mount(
      <Pagination
        currentPageNumber={1}
        totalNumberOfPages={10}
        onClick={() => {}}
      />
    );

    expect(wrapper.find('li.page-item').length).toBe(10);
  });
});
