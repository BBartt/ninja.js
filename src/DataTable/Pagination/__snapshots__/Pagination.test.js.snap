// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches snapshot 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      1
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link button-outline"
      onClick={[Function]}
    >
      2
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      3
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      4
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      5
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      6
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      7
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      8
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      9
    </button>
  </li>
  <li
    className="page-item mr-1"
  >
    <button
      className="page-link"
      onClick={[Function]}
    >
      10
    </button>
  </li>
</ul>
`;
